### DISCUSSION TITLE
Acorn BBC Micro Mode 7 (viewdata) support for BBS Boards

### DISCUSSION THREAD

**Discussion by Vela025** [mkrueger/icy_term#72] [Ideas]
I've set up a BBS running on my Acorn BBC Master which runs OBBS (1984), on a real BBC Micro this displays the colour codes ([as can be seen here](http://beebs.ddns.net/screenshots.html)), where as if you're accessing via a vt100 or ANSI terminal OBBS strips out these control codes and just displays the plain text in 40 column mode so that it is compatible with both.  Could support be added to Icy Term to allow it to show the colour codes.  I'm yet to find a terminal emulator that would support this but Icy Term seems like a good fit (already having vt100 and Viewdata support).
![SendingEm](https://github.com/mkrueger/icy_term/assets/155775589/954dc8f6-6828-44b4-b34d-2df18f515fe4)
BBS Mode 7 Board on BBC Micro
![ANSI EMAIL](https://github.com/mkrueger/icy_term/assets/155775589/21af7fb9-1d73-43f2-9aa0-1262c00912d1)
BBS Mode 7 Board on Syncterm.

**Comment by mkrueger** [comment]
Will look into it when it's online again. ATM I'm working on adding RIP/IGS support.

**Reply by mkrueger** [reply]
Ok looks doable it's a less complex viewdata like mode from what I can tell - I'll look what I can do.

I make first little progress for Mode7 support :)

![grafik](https://github.com/mkrueger/icy_term/assets/341098/e2ee3ef6-d477-4af6-a265-5e979148fbdc)


But I can really need some mode7 test files for completing that. Do you have any to share or a source where some are available? Or is there any program which can paint mode7 files?

**Comment by mkrueger** [comment]
A very basic preview mode7 is in 0.7.4 btw. but I need to test more :/.

**Comment by Vela025** [comment]
Oh thats great! I'll give it a whirl.

Cheers :)

Sent with [Proton Mail](https://proton.me/) secure email.
